### PR TITLE
Migrate Look & Feel Settings Page to Server-side

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
## What does this PR do?
- Migrates the component to server-side
- Uses server actions
- Adds loading screen
- Follows 1 component per file convention
- Adds `updateProductLookAndFeel` server action
- Introduces a zod input type for the above action

The server action is created specifically for this only and if needed, can be made generalised to edit other values of  the  product as well! I kept it specific to this for now since we only need it here and whenever we want to expand, we can do it accordingly!

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## Checklist
- [ ] Added a screen recording or screenshots to this PR
- [ ] Filled out the "How to test" section in this PR
- [x] Read the [contributing guide](https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/web/app/(app)/environments/[environmentId]/settings/lookandfeel/EditBrandColor.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      setUpdatingColor(false);
+    }
+  };
+
+  return (
+    <div className="w-full max-w-sm items-center">
+      <Label htmlFor="brandcolor">Color (HEX)</Label>
+      <ColorPicker color={color} onChange={setColor} />
+      <Button
+        type="submit"
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/settings/lookandfeel/EditBrandColor.tsx:34]: as it isn't a form that gets submitted, please remove the ` type="submit"`

File: apps/web/app/(app)/environments/[environmentId]/settings/lookandfeel/EditHighlightBorder.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import { useState } from "react";
+import toast from "react-hot-toast";
+import { DEFAULT_BRAND_COLOR } from "@formbricks/lib/constants";
+import { TProduct } from "@formbricks/types/v1/product";
+import { updateProduct } from "@formbricks/lib/services/product";
+
+export const EditHighlightBorder = ({ product }: { product: TProduct }) => {
+  const [showHighlightBorder, setShowHighlightBorder] = useState(product.highlightBorderColor ? true : false);
+  const [color, setColor] = useState<string | null>(
+    product.highlightBorderColor ? product.highlightBorderColor : DEFAULT_BRAND_COLOR
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/settings/lookandfeel/EditHighlightBorder.tsx:14]: would be cleaner using `product.highlightBorderColor || DEFAULT_BRAND_COLOR`

File: apps/web/app/(app)/environments/[environmentId]/settings/lookandfeel/EditHighlightBorder.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+        {showHighlightBorder && color ? (
+          <>
+            <Label htmlFor="brandcolor">Color (HEX)</Label>
+            <ColorPicker color={color} onChange={setColor} />
+          </>
+        ) : null}
+
+        <Button
+          type="submit"
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/settings/lookandfeel/EditHighlightBorder.tsx:64]: please also remove the `type="submit"` here

File: apps/web/app/(app)/environments/[environmentId]/settings/lookandfeel/EditPlacement.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+const placements = [
+  { name: "Bottom Right", value: "bottomRight", disabled: false },
+  { name: "Top Right", value: "topRight", disabled: false },
+  { name: "Top Left", value: "topLeft", disabled: false },
+  { name: "Bottom Left", value: "bottomLeft", disabled: false },
+  { name: "Centered Modal", value: "center", disabled: false },
+];
+
+export function EditPlacement({ product }: { product: TProduct }) {
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/settings/lookandfeel/EditPlacement.tsx:20]: Please also use an interface for the props here 😊

File: packages/lib/services/product.ts [hunk 5]
```diff
@@ -57,3 +58,35 @@ export const getProductByEnvironmentId = cache(async (environmentId: string): Pr
...
+      where: {
+        id: productId,
+      },
+      data: {
+        ...inputProduct,
+      },
+      select: selectProduct,
+    });
+    if (!updatedProduct) {
+      throw new ResourceNotFoundError("Update for Product", productId);
```
Comment (mattinannt) [inline packages/lib/services/product.ts:75]: This couldn't happen as Prisma would throw an error if the ressource that should be updated wasn't found.

**Review by mattinannt — CHANGES_REQUESTED** [review]
@ShubhamPalriwala thanks for the rewrite 😊 Structure looks good and I like it much better with no specific new type added. The services and types are like APIs which should be very generic and usable for different use cases.

I made a few suggestions about things I would change. They aren't necessarily things you programmed and could also be things that have been in the file before, but when we take the time to rewrite and revisit this page we can help to make it more consistent with the rest of the app 😊

File: apps/web/app/(app)/environments/[environmentId]/settings/lookandfeel/page.tsx
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/settings/lookandfeel/page.tsx:1]: Please add `export const revalidate = REVALIDATION_INTERVAL;` to make sure next.js doesn't cache indefinitely.

**Comment by ShubhamPalriwala** [comment]
Hey @mattinannt, thanks for the detailed feedback and apologies for missing the details here! Just made all the requested changes! Please take a look!

**Review by mattinannt — APPROVED** [review]
@ShubhamPalriwala Looks great now, thanks a lot 😊💪🎉
The only thing I changed was the order of the attributes for updateProduct to make it more consistent with the other services (ressourceId first, then new object) 😊

I will merge the changes into main now! 🎉🚀

**Pull request merged**